nomos-da/network/subnetworks-assignations/src/lib.rs

```diff
@@ -52,6 +56,8 @@ pub trait MembershipHandler {
 
     /// Returns all subnetworks with assigned members.
     fn subnetworks(&self) -> SubnetworkAssignations<Self::NetworkId, Self::Id>;
+
+    fn addressbook(&self) -> HashMap<Self::Id, Multiaddr>;
```

Comment:
This should probably be a reference. Or if we want to clone it for whatever reason an `Arc`, it makes no sense to return something that should be inmutable and just for reference.

Reply:
Membership is behind `ArcSwap`. I am not sure if we can return a reference in the thread safe data structures. Even if it was behind a regular `Mutex` we would still have a same problem I feel. 

But this method should be called only on update. Does that makes sense?

Reply:
Ok, we can clone it for now I guess. I remember we already had this convo anyway :sweat_smile:

Reply:
Yes, I think so as well. I still don't have a better approach, but I will think about it. I am not sure we can avoid clone because we are actually saving it to storage. This is where we use this call. Let's see :)